src/models/SAFT/PCSAFT/variants/PPCSAFT.jl

```diff
@@ -73,41 +80,191 @@ function a_res(model ::PPCSAFTModel, V, T, z)
 end
 
 function a_polar(model ::PPCSAFTModel, V, T, z, _data=@f(data))
-    dipole = model.params.dipole.values
-    if all(iszero,dipole)
-        return zero(V+T+first(z))
-    end
-    a₂ = @f(a_2,_data)
+    μ̄² = model.params.dipole2.values
+    Q̄² = model.params.quadrupole2.values
+    has_dp = !all(iszero, μ̄²)
+    has_qp = !all(iszero, Q̄²)
+    if !has_dp && !has_qp return 0. end

```

Comment:
in general, i recommend trying to use a pattern like the following:
```
_0 = zero(V+T+first(z))
if !has_dp && !has_qp return _0  end
```
0.0 is a `Float64`, and that could correspond to the type of the value when using `eos(model,V,T,z)`, but for other properties, AD is involved, and that zero could actually be a `ForwardDiff.Dual` instead